### ISSUE TITLE
Weird behaviour with lemonsqueezy script from registry

### ISSUE THREAD

**Issue by fayazara** [nuxt/scripts#98]
I am using the new nuxt/scripts module and a little confused.  I want to enable the lemonsqueezy script globally, I have multiple pages where this is needed. How should I do this? the [docs](https://scripts.nuxt.com/scripts/payments/lemon-squeezy#example) tell me to do the below and I added this in my `app.vue`

```html
<script setup>
const { Setup } = useScriptLemonSqueezy();
onMounted(() => {
  Setup();
});
</script>
```

Whats happening

1. `index.vue` pages opens the checkout overlay
2. `[...productId].vue` pages open the checkout overlay if I open the page directly, but not when I navigate to this page from a `<NuxtLink />` 

I thought this was an issue with the lemon.js script, but when I navigate to the product page from a NuxtLink, I can see `window.LemonSqueezy` has a been initiated which is confusing me even further, everything is working from both parties yet the checkout wont open in the overlay.

Here's a repro - https://stackblitz.com/edit/github-zk9jpe?file=pages%2Findex.vue

**Comment by harlan-zw** [comment]
Thanks for the details, I think this is what I also ran into when trying to implement the button (https://github.com/nuxt/scripts/issues/41). Will try and find the issue tomorrow.

**Comment by harlan-zw** [comment]
This should be fixed in v0.6.0, when using the composable you need to call refresh when the page updates as the Lemon.js library isn't reactive.

I've introduced the `<ScriptLemonSqueezy>` component as an alternative which takes care of this, see the [updated Lemon Squeezy docs](https://scripts.nuxt.com/scripts/payments/lemon-squeezy). 

Let me know if you have any feedback.
